napari/_qt/menus/_tests/test_file_menu.py

```diff
@@ -31,46 +36,119 @@ def _(path):
         uri='some-path/some-file.tif',
     )
     tmp_plugin.manifest.contributions.sample_data = [my_sample]
-
-    viewer = make_napari_viewer()
-    sample_action = viewer.window.file_menu.open_sample_menu.actions()[0]
+    app = get_app()
+    # required so setup steps run in init of `Viewer` and `Window`
+    make_napari_viewer()
     with mock.patch(
-        'napari._qt.menus.file_menu.handle_gui_reading'
+        'napari._qt.dialogs.qt_reader_dialog.handle_gui_reading'
     ) as mock_read:
-        sample_action.trigger()
+        app.commands.execute_command('tmp_plugin.tmp-sample')
 
     # assert that handle gui reading was called
     mock_read.assert_called_once()
 
 
 def test_plugin_display_name_use_for_multiple_samples(
-    make_napari_viewer, builtins
+    make_napari_viewer,
+    builtins,
 ):
-    """For plugin with more than two sample datasets, should use plugin_display for building the menu"""
+    """Check 'display_name' used for submenu when plugin has >1 sample data."""
+    app = get_app()
+    # Note this is required so `init_qactions` is run
```

Comment:
Here and in a couple of places we have this note about `make_napari_viewer` being run so `init_qactions` is run, but it's more than just `init_qactions`, it's about setting up all the appropriate menus, actions and callbacks. I would either remove this comment or expand it to say something like "we need the whole application configured and all actions registered for this test"

Reply:
I checked, in this case yes. I will amend.

I don't think all tests require everything e.g., I think one test previously just manually did `init_qactions` which I changed to `make_napari_viewer` because we decided that should always be used instead. My intent was to give an idea of what aspect of `make_napari_viewer` we actually needed for a test, that's why they are different. But potentially more is needed in the tests than I realised.

Reply:
I'm just going to add this comment to places where we don't need the viewer but need to run `make_napari_viewer`.